app/src/main/res/menu/menu_remove_search_engines.xml

```diff
@@ -8,5 +8,6 @@
     <item android:id="@+id/menu_delete_items"
         android:title="@string/preference_search_remove"
         android:icon="@drawable/ic_delete"
-        app:showAsAction="always"/>
+        app:showAsAction="always"
+        android:enabled="false" />
```

Comment:
Actually, if you just remove this "false" then the flicker goes away. What's happening is that the `invalidate` is recreating the menu each time, and setting it false, and then the code is re-enabling it, so we're seeing a flicker.

Reply:
+1